### ISSUE TITLE
Rule Change[no-unused-vars]: reportUsedIgnorePattern should check exported variable names

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#18728]
### What rule do you want to change?

no-unused-vars

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
export let _value = 0;
```

### What does the rule currently do for this code?

Does not report with option `reportUsedIgnorePattern: true` ([playground](https://eslint.org/play/#eyJ0ZXh0IjoiZXhwb3J0IGxldCBfdmFsdWUgPSAwO1xuXG5leHBvcnQgZnVuY3Rpb24gX2ZuKCkge31cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiLHsidmFyc0lnbm9yZVBhdHRlcm4iOiJeXyIsInJlcG9ydFVzZWRJZ25vcmVQYXR0ZXJuIjp0cnVlfV19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==))

### What will the rule do after it's changed?

Also check the names of directly exported variable, class and function declarations as they should also be counted as "used".

Maybe also include re-export renames:
```javascript
export { named as _named } from "mod";
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
In my opinion, enforcing naming conventions by this rule is fine for internally used variables, but for exported variables it seems out of the scope of its responsibilities. So I think that `reportUsedIgnorePattern: true` should ignore exported variables, as it is now.

@eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
I agree with @mdjermanovic. We don't actually know if an exported variable is ever used, so at this point we're really just asking to enforce a naming convention, which is beyond the scope of what the rule does.

**Comment by DMartens** [comment]
But this makes the case of direct vs indirect exports inconsistent:
```
// Reported as "used" even though only exported
function _f() {}

export {  _f }

// Currently not reported
export function _g() {}
```
